ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/InternalPaywallView.kt

```diff
@@ -20,50 +27,63 @@ import com.revenuecat.purchases.ui.revenuecatui.templates.Template3
 
 @Composable
 internal fun InternalPaywallView(
-    mode: PaywallViewMode = PaywallViewMode.default,
-    offering: Offering? = null,
-    listener: PaywallViewListener? = null,
-    viewModel: PaywallViewModel = getPaywallViewModel(offering = offering, listener = listener, mode = mode),
+    options: PaywallViewOptions,
```

Comment:
Note how I've changed from passing each of these separately to passing them all as part of the `PaywallViewOptions`. Since we need most of these from the view model anyway, I think this is cleaner.

Reply:
Nice good refactor.